### PR TITLE
Add Join Table For Visits

### PR DESCRIPTION
**Pull request by AyoubM490**
In this milestone, I:
- Added Join Table For Visits

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -42,4 +42,26 @@ ALTER TABLE animals ADD CONSTRAINT fk_animals_table FOREIGN KEY(species_id) REFE
...
+CREATE TABLE specializations(
+  SELECT *
+  FROM species 
+  FULL JOIN vets ON species.key = vets.key
+)
+CREATE TABLE visits(
+  SELECT *
+  FROM animals
+  FULL JOIN vets ON animals.key = vets.key
+)
```
Comment (DeliceLydia) [inline schema.sql]: - [x] This table is supposed to have `animal_id`, `vet_id`, `visit_date` columns

File: schema.sql [hunk 2]
```diff
@@ -42,4 +42,26 @@ ALTER TABLE animals ADD CONSTRAINT fk_animals_table FOREIGN KEY(species_id) REFE
...
+CREATE TABLE vets(
+  id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  name VARCHAR(50),
+  age INT,
+  date_of_graduation INT
+)
+CREATE TABLE specializations(
+  SELECT *
+  FROM species 
+  FULL JOIN vets ON species.key = vets.key
```
Comment (DeliceLydia) [inline schema.sql]: - [x] This table should have `species_id`, `vet_id` columns

File: data.sql [hunk 3]
```diff
@@ -55,4 +55,63 @@ UPDATE animals SET owner_id = (SELECT id FROM owners WHERE full_name = 'Bob') WH
...
+-- Boarmon visited Maisy Smith on May 15th, 2019.
+INSERT INTO visits (animal_id, vet_id, date_of_visit) VALUES (4, 2, '2019-05-15');
+-- Boarmon visited Maisy Smith on Feb 27th, 2020.
+INSERT INTO visits (animal_id, vet_id, date_of_visit) VALUES (4, 2, '2020-02-27');
+-- Boarmon visited Maisy Smith on Aug 3rd, 2020.
+INSERT INTO visits (animal_id, vet_id, date_of_visit) VALUES (4, 2, '2020-08-03');
+-- Blossom visited Stephanie Mendez on May 24th, 2020.
+INSERT INTO visits (animal_id, vet_id, date_of_visit) VALUES (6, 3, '2020-05-24');
+-- Blossom visited William Tatcher on Jan 11th, 2021.
+INSERT INTO visits (animal_id, vet_id, date_of_visit) VALUES (6, 1, '2021-01-11');
```
Comment (DeliceLydia) [inline data.sql]: - [x] Please add queries to answer the following questions
- Who was the last animal seen by William Thatcher?
- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- How many times were visits where with a vet that did not specialize in that animal's species?
- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @AyoubMs, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: schema.sql (hunk 1 above)
Comment (AyoubM490) [reply schema.sql]: ok

File: schema.sql (hunk 2 above)
Comment (AyoubM490) [reply schema.sql]: ok

File: data.sql (hunk 3 above)
Comment (AyoubM490) [reply data.sql]: Ok

File: schema.sql (hunk 1 above)
Comment (elisha2kyakpo1) [reply schema.sql]: - [x] Let's work on the requested changes.

File: schema.sql (hunk 2 above)
Comment (elisha2kyakpo1) [reply schema.sql]: - [x] kindly work on the changes requested by the previous reviewer

File: data.sql (hunk 3 above)
Comment (elisha2kyakpo1) [reply data.sql]: - [x] We should fix this change requested before submitting for another review

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi @AyoubMs, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (AyoubM490) [reply schema.sql]: ok

File: schema.sql (hunk 2 above)
Comment (AyoubM490) [reply schema.sql]: ok

File: data.sql (hunk 3 above)
Comment (AyoubM490) [reply data.sql]: ok

**Review by Whoistolu — COMMENTED** [review]
Hi @AyoubMs 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no changes. Concerning the last recommended changes by the previous reviewers kindly implement the changes before requesting another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @AyoubMs,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

You can do it

 ## Required Changes ♻️

- [x] REQUIRED:  Great implementation on the previous reviewer's requests on your tables.  Now, you should work on adding the required queries, as requested by previous reviewers:

- Who was the last animal seen by William Thatcher?
- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- How many times were visits where with a vet that did not specialize in that animal's species?
- What specialty should Maisy Smith consider getting? Look for the species she gets the most

These should go into `queries.sql`.  If you don't know how to implement these queries, please refer to the previous lessons.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 4]
```diff
@@ -1,108 +1,174 @@
...
+JOIN visits ON vets.id = visits.vet_id
+JOIN animals ON animals.id = visits.animal_id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(*) as num_animals_visited from vets
+	JOIN visits ON vets.id = visits.vet_id
+	WHERE vets.name = 'Stephanie Mendez';
```
Comment (Meltrust) [inline queries.sql]: - [x] This one is not selecting the different animals.  You need to use the keyword `distinct` like this:

```sql
SELECT DISTINCT animals.name
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE vets.name = 'Stephanie Mendez';
```

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @AyoubMs,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

You can do it

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (AyoubM490) [reply queries.sql]: ok

**Review by julie-ify — APPROVED** [review]
Hi @AyoubMs, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**